### ISSUE TITLE
 nfs-ganesha crashes with Segmentation fault

### ISSUE THREAD

**Issue by rakesh-roshan** [nfs-ganesha/nfs-ganesha#751]
Ganesha version: 3.5.
Ntirpc : v3.5
VFS: Custom VFS

The error log

```
[New LWP 3811]
[New LWP 28057]
[New LWP 28067]
[New LWP 28048]
[New LWP 9286]
[New LWP 28068]
[New LWP 9216]
[New LWP 28042]
[New LWP 9295]
[New LWP 28041]
[New LWP 9285]
[New LWP 28070]
[New LWP 9299]
[New LWP 28040]
[New LWP 28071]
[New LWP 28039]
[New LWP 7834]
[New LWP 9305]
[New LWP 7836]
[New LWP 9218]
[New LWP 28069]
[New LWP 4194]
[New LWP 9298]
[New LWP 8890]
[New LWP 7833]
[New LWP 6142]
[New LWP 4090]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `bin/ganesha.nfsd -f etc/ganesha/ganesha.conf -p var/run/ganesha.pid -F'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007f1268edf8b7 in mdcache_handle_to_key (obj_hdl=0x29e76108, fh_desc=0x7f125c665830) at /src/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_handle.c:1166
```

The stacktrace using gdb

```
(gdb) bt
#0  0x00007f1268edf8b7 in mdcache_handle_to_key (obj_hdl=0x29e76108, fh_desc=0x7f125c665830) at /src/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_handle.c:1166
#1  0x00007f1268e37b69 in state_obj_value_hash_func (hparam=0xe1ba20, key=0x7f125c6663e0) at /src/src/SAL/nfs4_state_id.c:362
#2  0x00007f1268dfdafd in compute (ht=0xe1ba20, key=0x7f125c6663e0, index=0x7f125c665b34, rbt_hash=0x7f125c665b20) at /src/src/hashtable/hashtable.c:253
#3  0x00007f1268dfe55c in hashtable_getlatch (ht=0xe1ba20, key=0x7f125c6663e0, val=0x7f125c6663c0, may_write=true, latch=0x7f125c6663a0) at /src/src/hashtable/hashtable.c:490
#4  0x00007f1268e38a63 in nfs4_State_Del (state=0x2ab8c870) at /src/src/SAL/nfs4_state_id.c:752
#5  0x00007f1268e32598 in _state_del_locked (state=0x2ab8c870, func=0x7f1268f2fda0 <__func__.20737> "state_nfs4_state_wipe", line=658) at /src/src/SAL/nfs4_state.c:351
#6  0x00007f1268e346cf in state_nfs4_state_wipe (ostate=0x29e76378) at /src/src/SAL/nfs4_state.c:658
#7  0x00007f1268e28349 in state_wipe_file (obj=0x29e76108) at /src/src/SAL/state_misc.c:1310
#8  0x00007f1268efce7a in _mdcache_lru_unref (entry=0x29e760d0, flags=0, func=0x7f1268f4ff6f <__func__.23209> "mdcache_put", line=199) at /src/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_lru.c:2057
#9  0x00007f1268edb37f in mdcache_put (entry=0x29e760d0) at /src/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_lru.h:199
#10 0x00007f1268edfc4c in mdcache_put_ref (obj_hdl=0x29e76108) at /src/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_handle.c:1329
#11 0x00007f1268e3a925 in nfs4_Check_Stateid (stateid=0x29111280, fsal_obj=0x0, state=0x7f125c667870, data=0x29e767b0, flags=0, owner_seqid=0, check_seqid=false, tag=0x7f1268f48be0 "TEST_STATEID")
    at /src/src/SAL/nfs4_state_id.c:1157
#12 0x00007f1268eb419b in nfs4_op_test_stateid (op=0x291111e0, data=0x29e767b0, resp=0x29111340) at /src/src/Protocols/NFS/nfs4_op_test_stateid.c:92
#13 0x00007f1268e88ed5 in process_one_op (data=0x29e767b0, status=0x7f125c6681bc) at /src/src/Protocols/NFS/nfs4_Compound.c:920
#14 0x00007f1268e89e0f in nfs4_Compound (arg=0x29e77560, req=0x29e76d50, res=0x29e76560) at /src/src/Protocols/NFS/nfs4_Compound.c:1327
#15 0x00007f1268dde557 in nfs_rpc_process_request (reqdata=0x29e76d50) at /src/src/MainNFSD/nfs_worker_thread.c:1508
#16 0x00007f1268dde82f in nfs_rpc_valid_NFS (req=0x29e76d50) at /src/src/MainNFSD/nfs_worker_thread.c:1625
#17 0x00007f1268b65e04 in svc_vc_decode (req=0x29e76d50) at /src/src/libntirpc/src/svc_vc.c:904
#18 0x00007f1268b61cc8 in svc_request (xprt=0xe4a900, xdrs=0x29e74830) at /src/src/libntirpc/src/svc_rqst.c:1202
#19 0x00007f1268b65d0d in svc_vc_recv (xprt=0xe4a900) at /src/src/libntirpc/src/svc_vc.c:877
#20 0x00007f1268b61c49 in svc_rqst_xprt_task_recv (wpe=0xe4ab38) at /src/src/libntirpc/src/svc_rqst.c:1183
#21 0x00007f1268b627dd in svc_rqst_epoll_loop (wpe=0xe23128) at /src/src/libntirpc/src/svc_rqst.c:1564
#22 0x00007f1268b6d327 in work_pool_thread (arg=0xe6ff40) at /src/src/libntirpc/src/work_pool.c:184
#23 0x00007f1267b6fe75 in start_thread () from /lib64/libpthread.so.0
#24 0x00007f126726f8fd in clone () from /lib64/libc.so.6
```

The error is occuring at [mdcache_handle.c#L1166](https://github.com/nfs-ganesha/nfs-ganesha/blob/V3.5/src/FSAL/Stackable_FSALs/FSAL_MDCACHE/mdcache_handle.c#L1166-L1169)

**Comment by ffilz** [comment]
Daniel any thoughts on this one?

**Comment by dang** [comment]
Seems like the sub-handle is gone, but hard to know for sure.  It seems odd that we'd be in CLEANUP in this codepath, but it's not impossible.  Some questions:

How reproducible is this?
Are there logs?
Can you catch this in then debugger?

**Comment by rakesh-roshan** [comment]
It is intermittent. I have coredump file but no other logs. I have not yet figured out a deterministic way to reproduce it.

**Comment by ffilz** [comment]
With no way to reproduce this in over 30 days, I'm going to close this. Please re-open if you have more details that we can use.

Also, since we have now released V4.0, V3.x is technically unsupported now so a reproduction on V4.0 or later would be appreciated. Marking as try on supported version to reflect this.
